### PR TITLE
Non-trivial merge to main

### PR DESCRIPTION
**Pull request by rossabaker**
Some things (quicklint, scalafix disablement on root projects) were oddly missing.  I pieced it back together, but I'm scared.

### CODE REVIEW

File: .scalafix.conf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+rules = [
```
Comment (bplommer) [inline .scalafix.conf:1]: Looks like the newer changes have gone missing

Comment (rossabaker) [reply .scalafix.conf:1]: I noticed some things in build.sbt didn't make it.  I'm now scared of what else didn't.  I don't know what I've done wrong here.

Comment (rossabaker) [reply .scalafix.conf:1]: I see now.  I think my local series/0.23 wasn't fully up to date.

Comment (rossabaker) [reply .scalafix.conf:1]: Yep, that's it.  I should have chased my suspicions harder.

```
Head:     series/0.23 Merge pull request #5506 from http4s/non-trivial-merge-2020-10-27
Merge:    origin/series/0.23 Merge pull request #5529 from http4s/non-trivial-merge-2021-11-02
Push:     origin/series/0.23 Merge pull request #5529 from http4s/non-trivial-merge-2021-11-02
```

**Comment by rossabaker** [comment]
One build got killed for memory.  Not sure if that's my fault or not.

**Comment by armanbilge** [comment]
Unusual, the JS jobs have been pretty reliable I think. But nothing seems obviously wrong.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,6 +36,18 @@ jobs:
...
+            java: adoptium@11
+          - ci: ciJVM
+            scala: 2.12.15
+            java: adoptium@17
+          - ci: ciJVM
+            scala: 3.1.0
+            java: adoptium@11
+          - ci: ciJVM
+            scala: 3.1.0
+            java: adoptium@17
```
Comment (bplommer) [inline .github/workflows/ci.yml:50]: Should these be re-thinned?

Comment (rossabaker) [reply .github/workflows/ci.yml:50]: Yes.  I didn't notice they were thickened.

Comment (rossabaker) [reply .github/workflows/ci.yml:50]: Wait, is it wrong?  It's running the JVM on each Java for 2.13, on each Scala for Adoptium 8.  And it's running the Node build once each Scala.  And then duplication from pull_request / push.

Comment (bplommer) [reply .github/workflows/ci.yml:50]: Not sure, but shouldn’t it be the same in 0.23 and main? (cc @armanbilge )

Comment (armanbilge) [reply .github/workflows/ci.yml:50]: Sorry I looked too quickly when I 👍 yesterday, retracted 😉 

This looks correct. It's just confusing (at least for me), b/c the list @bplommer commented on is the _excluded_ job list. If you look below at what CI actually did it's right.

Comment (rossabaker) [reply .github/workflows/ci.yml:50]: So this one is good then?  There's a chonky one coming up behind it.

**Review by armanbilge — APPROVED** [review]

**Pull request merged**